default_app/main.js

```diff
@@ -234,40 +234,62 @@ function loadApplicationPackage (packagePath) {
     packagePath = path.resolve(packagePath)
     const packageJsonPath = path.join(packagePath, 'package.json')
     if (fs.existsSync(packageJsonPath)) {
-      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath))
+      let packageJson
+      try {
+        packageJson = JSON.parse(fs.readFileSync(packageJsonPath))
+      } catch (e) {
+        showErrorMessage('Unable to parse package.json, it contains errors.\n\n' +
+          `${e.toString()} in ${packageJsonPath}`)
+      }
       if (packageJson.version) app.setVersion(packageJson.version)
 
       if (packageJson.productName) {
         app.setName(packageJson.productName)
       } else if (packageJson.name) {
         app.setName(packageJson.name)
       }
-
+      if (!packageJson.main) {
```

Comment:
I think `main` is optional and defaults to `index.js` when not present so we might not want this check here and just catch the `main` being an invalid path down below where the package path is resolved.